transport-native-epoll/src/main/java/io/netty/channel/epoll/TcpMd5Util.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.epoll;
+
+import java.net.InetAddress;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+
+final class TcpMd5Util {
+
+    private TcpMd5Util() {
+    }
+
+    static Set<InetAddress> setTcpMd5Sig(AbstractEpollChannel channel, Set<InetAddress> current,
+                                         Map<InetAddress, byte[]> newKeys) {
+        // Validate incoming values
+        for (Entry<InetAddress, byte[]> e : newKeys.entrySet()) {
+            final byte[] key = e.getValue();
+            if (e.getKey() == null) {
+                throw new IllegalArgumentException("Null address");
+            }
+            if (key == null) {
+                throw new IllegalArgumentException("Null key for address " + e.getKey());
+            }
+            if (key.length == 0) {
+                throw new IllegalArgumentException("Empty key for address " + e.getKey());
+            }
+            if (key.length > Native.TCP_MD5SIG_MAXKEYLEN) {
+                throw new IllegalArgumentException("Key length for address " + e.getKey() +
```

Comment:
`"newKeys[" + e.getKey() + "] has a key with invalid length; should not exceed " + TCP_MD5SIG_MAXKEYLEN`